### ISSUE TITLE
tofo init cannot find module version with a suffix

### ISSUE THREAD

**Issue by c-baumgartner** [opentofu/registry#1094]
### Thank you for reporting an issue.

Trying to use module `Azure/avm-ptn-alz/azurerm` I have a strange behavior. Without a version constraint it will always use version `0.8.1`

Trying specifing the version manually to `v0.9.0-beta2` will result in the following error:

```text
╷
│ Error: Unresolvable module version constraint
│ 
│   on main.tf line 8:
│    8: module "alz_architecture" {
│ 
│ There is no available version of module "registry.opentofu.org/Azure/avm-ptn-alz/azurerm" (main.tf:8) which matches the given version constraint. The newest available version is 0.8.1.
╵
```

Looking into the `azurerm.json` in the module catalog the beta-versions are available (only an excerpt from the file):

```json
{
  "versions": [
    {
      "version": "v0.9.0-beta2"
    },
    {
      "version": "v0.9.0-beta"
    },
    {
      "version": "v0.8.1"
    }]
}


### OpenTofu Version

```shell
OpenTofu v1.8.2
on linux_amd64
```

### OpenTofu Configuration Files

```hcl
module "alz_architecture" {
  source  = "Azure/avm-ptn-alz/azurerm"
  version = "v0.9.0-beta"

  architecture_name  = "alz"
  parent_resource_id = data.azapi_client_config.current.tenant_id
  location           = local.location
}
```

**Comment by ollevche** [comment]
Hey @c-baumgartner thank you for the issue! I believe you shouldn't specify the `v` prefix in module version so `version = "v0.9.0-beta"` becomes `version = "0.9.0-beta"`. However, I agree error message doesn't make it clear at all.

**Comment by c-baumgartner** [comment]
Hi @ollevche - thank you for pointing that out - fun fact: Using "v0.8.1" ist working without any issue. It only fails for the version with the "-beta" suffix. The versions in the azurerm.json file are using the "v" too :-) So yeah, really confusing. But without the "v" it is working also for the -beta version constraint....

**Comment by ollevche** [comment]
@c-baumgartner I agree, that's inconsistent. We will discuss this issue at the core team triage meeting and hopefully change the error message or parsing logic.
